### PR TITLE
Remove deprecated legacy color names

### PR DESCRIPTION
**Pull request by amelako**
## Purpose

Remove the deprecated legacy color names (`success`, `warning` and `danger`)

## Approach and changes

- remove the deprecated `success`, `warning` and `danger` color names

- replace its use in the Body and Badge components

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: .changeset/dull-rats-fry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/design-tokens': major
+---
+
+Removed the deprecated `success`, `warning` and `danger` legacy color names. Use semantic colors `confirm`, `notify` and `alert`.
```
Comment (robinmetral) [inline .changeset/dull-rats-fry.md:5]:
```suggestion
Removed the deprecated `success`, `warning` and `danger` color variants. Use `confirm`, `notify` and `alert` instead.
```

File: .changeset/dull-rats-fry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/design-tokens': major
```
Comment (robinmetral) [inline .changeset/dull-rats-fry.md:2]: You also need another changeset for `@sumup/design-tokens` 🙂

File: packages/circuit-ui/components/ListItem/ListItem.spec.tsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ describe('ListItem', () => {
       const wrapper = renderListItem(create, {
         ...baseProps,
         prefix: (
-          <Badge variant="danger" circle>
+          <Badge variant="alert" circle>
```
Comment (robinmetral) [inline packages/circuit-ui/components/ListItem/ListItem.spec.tsx:68]: Thanks for also migrating other stories!

**Review by robinmetral — COMMENTED** [review]
Nice!

**Review by connor-baer — APPROVED** [review]
Nothing to add to Robin's comments.

**Pull request merged**